### DISCUSSION TITLE
Swapping Body Texture (Diffuse) getting CORS

### DISCUSSION THREAD

**Discussion by WadeMorrison** [google/model-viewer#4134] [Q&A]
Hey there, was hopping to see if anyone could help me out here. I have a configurator that is able to swap materials/textures instead of variants. And when I click on any of the materials, it'll "GET" and swap the ORM and Normal textures, but not the Diffuse? All textures live in the same bucket with the correct CORS settings on AWS. My app works great locally, but when deployed to Heroku it gets the error. I can swap models/.GLBs as well with no problem. It's only with swapping the diffuse channel. 

I have cors middleware on my backend as well. 

Hope someone can help! I can also provide a link to an example if that helps.

**Comment by WadeMorrison** [comment]
Messed around with a few more items, but no luck. I changed my CORS setting to '*' on my server.js instead of my app's url. And still not working. Again, it's "GET"ting the normal and orm maps just fine, and even the diffuse for the material swatch is rendering without issue. In the Network pane, it shows it's not able to get it once I swap the materials.

Here's an example for the configurator: 
https://fathomless-brook-11892.herokuapp.com/product/63ee919d0c2221a7a066c9f7

**Reply by elalish** [reply]
Hmm, it didn't work for me the first time either, but upon reload (at least with dev tools open) it started working fine. Maybe you just solved it?

**Reply by WadeMorrison** [reply]
Oh interesting, i haven’t pushed any new updates, but I didn’t try refreshing it and trying again. Should I add a timeout to the material swap maybe? I’ll try loading it and refreshing the page tonight and see if it works for me as well.

**Reply by elalish** [reply]
CORS errors have nothing to do with MV, so I'd guess it's something about your server taking time to update.

**Reply by WadeMorrison** [reply]
Yeah that’s I figured, I just assumed that since I can GET the image for swatch <label>  with no issue, but it getting tripped up on the diffuse texture swap was maybe MV issue.

 I’ll keep messin with it and report back if I find anything! Appreciate you lookin into it!
